roborock/devices/b01_channel.py

```diff
@@ -58,20 +61,24 @@ def find_response(response_message: RoborockMessage) -> None:
             if isinstance(inner, dict) and inner.get("msgId") == msg_id:
                 _LOGGER.debug("Received query response: %s", inner)
                 data = inner.get("data")
+                # All get commands should be dicts
+                if command.endswith(".get") and not isinstance(data, dict):
+                    if not future.done():
+                        future.set_exception(RoborockException("Unexpected data type for response"))
+                    return
                 if not future.done():
-                    if isinstance(data, dict):
-                        future.set_result(data)
-                    else:
-                        future.set_exception(RoborockException(f"Unexpected data type for response: {data}"))
+                    future.set_result(data)
 
     unsub = await mqtt_channel.subscribe(find_response)
 
     _LOGGER.debug("Sending MQTT message: %s", roborock_message)
     try:
         await mqtt_channel.publish(roborock_message)
-        try:
-            return await asyncio.wait_for(future, timeout=_TIMEOUT)
-        except TimeoutError as ex:
-            raise RoborockException(f"Command timed out after {_TIMEOUT}s") from ex
+        return await asyncio.wait_for(future, timeout=_TIMEOUT)
+    except TimeoutError as ex:
+        raise RoborockException(f"Command timed out after {_TIMEOUT}s") from ex
+    except Exception as ex:
+        _LOGGER.exception("Error sending decoded command: %s", ex)
```

Comment:
Catch `RoborockException` and raise? Catching bare exception and logging exception should be for unexpected cases only